### DISCUSSION TITLE
Fabric Canvas drawing in Webview causes 100% CPU

### DISCUSSION THREAD

**Discussion by ouya99** [fabricjs/fabric.js#9541] [General]
Hi,

i tried to run a fabricjs canvas (v5.30) in my webview and it always will drive CPU to 100% very shortly after i start to draw continuously

1. RN App with react-native-webview": "13.2.2" (launches <WebView )
2. Webview holds most basic fabricjs canvas with isDrawingMode = true
3. Draw for many seconds without pause (after 10-15 seconds CPU will show 100%)

Anyone with this issue?

Also does anyone have a very basic sample for v6 to test this again in new fabric v6 ? Thanks

**Comment by ShaMan123** [comment]
RN + webview is very bad in perf the last I worked with it (which was a long time ago)
RN + native canvas is even not good if I recall correctly.
So I am not sure it is the repo's scope.
However, I suggest you try a native drawing tool and see if that doesn't block your CPU before moving forward.
v6 code samples can be found at the bottom of this [page](https://github.com/fabricjs/fabric.js/issues/new/choose)
![image](https://github.com/fabricjs/fabric.js/assets/34343793/84d04ab8-7fe3-49fe-9bfe-950285ff0295)

**Comment by ouya99** [comment]
```
        const options = {isDrawingMode: true, backgroundColor: 'orange', width:300, height:300};
        const canvas = new fabric.Canvas(canvasEl.current, options);
```

@ShaMan123 
tried this but i am not even sure how _options_ will be used? Any API on that. it kind of worked with v5 but does not with v6

**Reply by ShaMan123** [reply]
https://codesandbox.io/p/devbox/fabric-vanillajs-sandbox-forked-ycrl3q?file=%2Fsrc%2Findex.ts%3A10%2C29

**Reply by ouya99** [reply]
tried to run https://github.com/fabricjs/ reactJS- but seems not to load fabricjs. Does it work for both versions v5 and v6 ?

**Reply by ShaMan123** [reply]
That repo is abandoned

**Reply by ShaMan123** [reply]
wait, I am not sure I understand

**Reply by ouya99** [reply]
tried to run fabricjs in reactJS. works with v5, but not sure how to implement it in v6. what is abandoned?

**Reply by ShaMan123** [reply]
The reactjs repo is abandoned
Stick to the codesandbox
